src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -803,38 +803,79 @@ public BoundSequence Sequence(ImmutableArray<LocalSymbol> locals, ImmutableArray
             return new BoundSequence(Syntax, locals, sideEffects, result, result.Type) { WasCompilerGenerated = true };
         }
 
+        /// <summary>
+        /// An internal helper class for building a switch statement.
+        /// </summary>
+        internal class SyntheticSwitchSection
+        {
+            public readonly ImmutableArray<int> Values;
+            public readonly ImmutableArray<BoundStatement> Statements;
+            public SyntheticSwitchSection(ImmutableArray<int> Values, ImmutableArray<BoundStatement> Statements)
+            {
+                this.Values = Values;
+                this.Statements = Statements;
+            }
+        }
+
+        public SyntheticSwitchSection SwitchSection(int value, params BoundStatement[] statements)
+        {
+            return new SyntheticSwitchSection(ImmutableArray.Create(value), ImmutableArray.Create(statements));
+        }
+
+        public SyntheticSwitchSection SwitchSection(List<int> values, params BoundStatement[] statements)
+        {
+            return new SyntheticSwitchSection(ImmutableArray.CreateRange(values), ImmutableArray.Create(statements));
+        }
+
         /// <summary>
         /// Produce an int switch.
         /// </summary>
-        public BoundStatement Switch(BoundExpression ex, params BoundSwitchSection[] sections)
+        public BoundStatement Switch(BoundExpression ex, params SyntheticSwitchSection[] sections)
```

Comment:
This `params` looks unnecessary. I think the only caller is below. Can we just make this ImmutableArray and force all the callers to properly construct their arguments? And delete the overload below?

Reply:
Yes; recorded in [https://github.com/dotnet/roslyn/issues/25663](https://github.com/dotnet/roslyn/issues/25663)

---
In reply to: [177231955](https://github.com/dotnet/roslyn/pull/25544#discussion_r177231955) [](ancestors = 177231955)